cloudberry/zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/OracleGenerator.scala

```diff
@@ -0,0 +1,408 @@
+
+package edu.uci.ics.cloudberry.zion.model.impl
+
+import edu.uci.ics.cloudberry.zion.model.datastore.{IQLGenerator, IQLGeneratorFactory, QueryParsingException}
+import edu.uci.ics.cloudberry.zion.model.schema._
+import org.joda.time.DateTime
+import play.api.libs.json._
+
+import scala.collection.mutable
+import scala.collection.mutable.{ArrayBuffer, ListBuffer}
+
+class OracleGenerator extends SQLGenerator {
+
+  protected val quote = '"'
+  protected val truncate: String = "truncate"
+  protected val fullTextMatch = Seq("contains", "")
+
+  //converts a value in internal geometry format to its plain text representation, e.g.: "POINT(1, 2)"
+  private val geoAsText: String = "st_astext"
+  private val pointGetCoord = Seq("st_x", "st_y")
+
+  override protected def genDDL(name: String, schema: Schema): String = {
+
+    def mkNestDDL(names: String, typeStr: String): String = {
+      names match {
+        case e => s"  $quote$e$quote $typeStr"
+      }
+    }
+    val fields = schema.fieldMap.values.filter(f => f.dataType != DataType.Hierarchy && f != AllField).map {
+      f => mkNestDDL(f.name, fieldType2SQLType(f) + (if (f.isOptional) " default null" else " not null"))
+    }
+    s"""
+       |declare
+       |    result1 number(8);
+       |begin
+       |    select count(*) into result1 from dba_tables where owner = 'BERRY' and table_name = '${name}';
+       |if result1 = 0 then
+       |execute immediate 'create table $quote${name}$quote (
+       |${fields.mkString(",\n")}, primary key (${schema.primaryKey.map(key => key.name).mkString(s"$quote",s"$quote,$quote",s"$quote")})
+       |)';
+       |end if;
+       |end;
+       |/\n""".stripMargin
+  }
+
+  override protected def parseSelect(selectOpt: Option[SelectStatement],
+                                     exprMap: Map[String, FieldExpr], query: Query,
+                                     queryBuilder: StringBuilder): ParsedResult = {
+    selectOpt match {
+      case Some(select) =>
+
+        val producedExprs = mutable.LinkedHashMap.newBuilder[String, FieldExpr]
+
+        val orderStrs = select.orderOn.zip(select.order).map {
+          case (orderOn, order) =>
+            val expr = exprMap(orderOn.name).defExpr
+            val orderStr = if (order == SortOrder.DSC) "desc" else ""
+            s"${expr} $orderStr"
+        }
+        val orderStr =
+          if (orderStrs.nonEmpty) {
+            orderStrs.mkString("order by ", ",", "")
+          } else {
+            s""
+          }
+        val limitStr = s"fetch first ${select.limit} rows only"
+        appendIfNotEmpty(queryBuilder, orderStr)
+        if (select.limit != 0) {
+          appendIfNotEmpty(queryBuilder, limitStr)
+        }
+        if (select.fields.isEmpty || query.hasUnnest || query.hasGroup) {
+          producedExprs ++= exprMap
+        } else {
+          select.fields.foreach {
+            field => {
+              if (field.dataType != DataType.Point) {
```

Comment:
Same concern, if `ojdbc` supports `SDO_GEOMETRY` type, we don't need to convert them into `VARCHAR2`.